tests/rptest/tests/transactions_test.py

```diff
@@ -489,6 +490,56 @@ def consume_records(consumer, count):
 
         consume_records(consumer, 200)
 
+    @cluster(num_nodes=3)
+    def check_pids_overflow_test(self):
+        rpk = RpkTool(self.redpanda)
+        ans = rpk.cluster_config_set("max_concurrent_producer_ids", str(10))
+        print(ans)
+
+        test_producer = producer = ck.Producer({
+            'bootstrap.servers':
+            self.redpanda.brokers(),
+            'enable.idempotence':
+            True,
+        })
+
+        topic = self.topics[0].name
+        test_producer.produce(topic,
+                              'test',
+                              'test',
+                              partition=0,
+                              on_delivery=self.on_delivery)
+        test_producer.flush()
+
+        producers = []
+        for i in range(50):
+            p = ck.Producer({
+                'bootstrap.servers': self.redpanda.brokers(),
+                'enable.idempotence': True,
+            })
+            p.produce(topic,
+                      str(i),
+                      str(i),
+                      partition=0,
+                      on_delivery=self.on_delivery)
+            p.flush()
+            producers.append(p)
+
+        # Need to wait, but we can not understand when cleaning will be finished in rm_stm
+        time.sleep(10)
```

Comment:
that's awkward that we need to sleep, maybe invoke `clear_old_idempotent_pids` from `replicate_seq` if we past the threshold and remove old pids before adding new in this case we don't need to sleep

Reply:
I think idea was remove in background.

Reply:
On leader I remove also in replicate_seq